app/src/features/map/MapPage.tsx

```diff
@@ -0,0 +1,129 @@
+import { Container, Grid, Typography } from '@material-ui/core';
+import Box from '@material-ui/core/Box';
+import { IErrorDialogProps } from 'components/dialog/ErrorDialog';
+import { IMarker } from 'components/map/components/MarkerCluster';
+import MapContainer from 'components/map/MapContainer';
+import { IGetMapOccurrenceData, OccurrenceFeaturePopup } from 'components/map/OccurrenceFeaturePopup';
+import { DialogContext } from 'contexts/dialogContext';
+import { Feature } from 'geojson';
+import { APIError } from 'hooks/api/useAxios';
+import { useApi } from 'hooks/useApi';
+import { LatLngBounds } from 'leaflet';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+
+const MapPage: React.FC = () => {
+  const platformApi = useApi();
+
+  const [performSearch, setPerformSearch] = useState<boolean>(true);
+  const [geometries, setGeometries] = useState<any[]>([]);
+
+  const dialogContext = useContext(DialogContext);
+
+  const showFilterErrorDialog = useCallback(
+    (textDialogProps?: Partial<IErrorDialogProps>) => {
+      dialogContext.setErrorDialog({
+        onClose: () => {
+          dialogContext.setErrorDialog({ open: false });
+        },
+        onOk: () => {
+          dialogContext.setErrorDialog({ open: false });
+        },
+        ...textDialogProps,
+        open: true
+      });
+    },
+    [dialogContext]
+  );
+
+  const getOccurrenceData = useCallback(
+    async (bounds?: LatLngBounds) => {
+      try {
+        let spatialBounds: Feature = {} as unknown as Feature;
+
+        if (bounds) {
+          const southWest = bounds.getSouthWest();
+          const northEast = bounds.getNorthEast();
+
+          spatialBounds = {
+            type: 'Feature',
+            properties: {},
+            geometry: {
+              type: 'Polygon',
+              coordinates: [
+                [
+                  [southWest.lng, southWest.lat],
+                  [southWest.lng, northEast.lat],
+                  [northEast.lng, northEast.lat],
+                  [northEast.lng, southWest.lat],
+                  [southWest.lng, southWest.lat]
+                ]
+              ]
+            }
+          } as Feature;
+        }
```

Comment:
Not critical, but might be nice to move this logic to a util function somewhere.